pilot/pkg/model/service.go

```diff
@@ -60,14 +60,20 @@ type Service struct {
 	Mutex sync.RWMutex
 	// ClusterVIPs specifies the service address of the load balancer
 	// in each of the clusters where the service resides
-	ClusterVIPs map[string]string `json:"cluster-vips,omitempty"`
+	ClusterVIPs map[string]string `json:"clusterVips,omitempty"`
+
+	// LoadbalancerAddresses is a mapping between a cluster name and the load
+	// balancer address(es) to access the service from outside the cluster.
+	// Used by the aggregator to aggregate the Attributes.LoadBalancerAddresses
+	// for clusters where the service resides
+	LoadbalancerAddresses map[string][]string
```

Comment:
this is not clean. This space is effectively a temporary storage for aggregated load balancer addresses.
Secondly, the concept of a load balancer address in the service is very strange when we have an address field specifically for this. Unlike kubernetes, in VM environments (behind hardware load balancers), the service address is the load balancer address. There is no internal or external address. Its just one.

Reply:
Come up with two names  `externalAddresses` `publicAddresses`, I prefer `externalAddresses`. And I will comment clearly.

in VM environments, we use consul, the external addr is selected by the order TaggedAddresses -> ServiceAddress->Address

Reply:
No need to consider consul, for vm there is no multi cluster concept.